### ISSUE TITLE
Report failure if file isn't successfully written

### ISSUE THREAD

**Issue by pshriwise** [svalinn/DAGMC-Trelis#150]
We've seen a couple of cases on OSX where the working directory of the Cubit application is `/` or some other location that isn't writeable for the user. In this case the plugin will either cause Cubit to crash or finish without writing the file, neither of which are desirable behavior if we fail to write the DAGMC file.

**Comment by MicahGale** [comment]
This error should occur ASAP IMO. A helpful error message and crash after a long execution is barely more useful than no error message at all. As soon as the file location is known something simulating `touch` should be done to verify write permissions.

**Comment by pshriwise** [comment]
Agreed. I'm betting there's a convention for handling this within Cubit either by detecting permissions of the output directory through the OS or writing a test file as you suggested.
